prow/tide/tide_test.go

```diff
@@ -1406,3 +1421,201 @@ func TestSync(t *testing.T) {
+			requiredContexts:      []string{"c1", "c2", "c3"},
+		},
+		{
+			name:                  "required checks but not used - failing",
+			passing:               false,
+			useRequiredChecksOnly: false,
+			combinedContexts:      map[string]string{"c1": success, "c2": failure},
+			requiredContexts:      []string{"c1", "c2", "c3"},
+		},
+		{
+			name:                  "required checks used - passing",
+			passing:               true,
+			useRequiredChecksOnly: true,
+			combinedContexts:      map[string]string{"c1": success, "c2": failure, "c3": success},
+			requiredContexts:      []string{"c1", "c3"},
+		},
+		{
+			name:                  "required checks used - failing",
+			passing:               false,
+			useRequiredChecksOnly: true,
+			combinedContexts:      map[string]string{"c1": success, "c2": success},
+			requiredContexts:      []string{"c1", "c2", "c3"},
+		},
+	}
+
+	for _, tc := range testCases {
+		ghc := &fgc{
+			requiredChecks: tc.requiredContexts,
+			combinedStatus: tc.combinedContexts,
+			expectedSHA:    headSHA}
+		log := logrus.WithField("component", "tide")
+		_, err := log.String()
+		if err != nil {
+			t.Errorf("Failed to get log output before testing: %v", err)
+			t.FailNow()
+		}
+		bp := branchProtection{
+			config:         config.TideBranchProtection{UseRequiredChecksOnly: tc.useRequiredChecksOnly},
+			requiredChecks: sets.NewString(tc.requiredContexts...),
+		}
+		pr := PullRequest{HeadRefOID: githubql.String(headSHA)}
+		passing := isPassingTests(log, ghc, pr, bp)
+		if passing != tc.passing {
+			t.Errorf("%s: Expected %b got %b", tc.name, tc.passing, passing)
+		}
+	}
+}
+
+func TestBranchProtectionSkipContext(t *testing.T) {
+	testCases := []struct {
+		name           string
+		requiredChecks []string
+		config         config.TideBranchProtection
+		context        string
+		skipContext    bool
+	}{
+		{
+			name:        "no config no required checks",
+			config:      config.TideBranchProtection{},
+			context:     "c1",
+			skipContext: false,
+		},
+		{
+			name:           "no config existing context",
+			requiredChecks: []string{"c1", "c2", "c3"},
+			config:         config.TideBranchProtection{},
+			context:        "c1",
+			skipContext:    false,
+		},
+		{
+			name:           "no config missing context",
+			requiredChecks: []string{"c1", "c2", "c3"},
+			config:         config.TideBranchProtection{},
+			context:        "c4",
+			skipContext:    false,
+		},
+		{
+			name:           "config existing context",
+			requiredChecks: []string{"c1", "c2", "c3"},
+			config: config.TideBranchProtection{
+				UseRequiredChecksOnly: true,
+			},
+			context:     "c1",
+			skipContext: false,
+		},
+		{
+			name:           "config missing context",
+			requiredChecks: []string{"c1", "c2", "c3"},
+			config: config.TideBranchProtection{
+				UseRequiredChecksOnly: true,
+			},
+			context:     "c4",
+			skipContext: true,
+		},
+		{
+			name: "config no required checks",
```

Comment:
This seems like something we should validate at config load time -- if you require checks but configured none, that's a user error in config

Reply:
the required checks are coming from the github api, so validating the config would require an api call for every branch, plus checking it at config parsing does not actually guarantees that it'll still be valid a couple of minute after.